ironfish/src/memPool/feeEstimator.ts

```diff
@@ -46,101 +53,118 @@ export class FeeEstimator {
       const currentBlock = await this.chain.getBlock(currentBlockHash)
       Assert.isNotNull(currentBlock, 'No block found')
 
-      const lowestFeeTransactions = this.getLowestFeeRateTransactions(
+      const percentileTransactions = this.getPercentileFeeRateTransactions(
         currentBlock,
-        this.numOfTxSamples,
+        this.percentiles,
       )
 
-      for (const transaction of lowestFeeTransactions) {
-        if (this.isFull()) {
-          break
+      this.percentiles.forEach((percentile, i) => {
+        const priorityLevel = PERCENTILES_TO_PRIORITY_LEVELS.get(percentile)
+
+        if (priorityLevel) {
+          const queue = this.queues.get(priorityLevel)
```

Comment:
I'm wondering if it makes sense here to iterate over the queues instead of the percentiles?

With the percentiles we need to look up the priority level and the queue and check that each is non-null. We could also include the percentile in the queues data structure so that it's a `Map<PriorityLevel, [Percentile, FeeRateEntry[]]>`

Reply:
I updated to iterate over the queues, it has fewer if conditions to check. I also changed the `getPercentileFeeRateEntry` to return a map of percentile and fee rate entry. So its easier to look up the fee rate entry to add to queue.